untie/src/untie.cpp

```diff
@@ -57,91 +60,119 @@ Untie::Untie() {
-        material.set_texture_slot_path(TextureType::Occlusion, "whiteTexture");
+        material.set_texture_slot_path(TextureType::Metallic, "normal_tiles_1k.png");
+        material.set_texture_slot_path(TextureType::Roughness, "normal_tiles_1k.png");
+        material.set_texture_slot_path(TextureType::Occlusion, "normal_tiles_1k.png");
         cubeObj.add_component<components::Material>(material);
     }
     {
         auto cubeObj = scene.create_game_object("loaded_dragon");
         cubeObj.get_component<components::Transform>().set_position(glm::vec3(-5.0f, 1.0f, -10.0f));
         cubeObj.get_component<components::Transform>().set_scale(glm::vec3(5, 5, 5));
         cubeObj.get_component<components::Transform>().set_rotation_euler(glm::vec3(0, 180, 0));
-        cubeObj.add_component<InstanceMesh>("dragon.obj");
+        cubeObj.add_component<components::InstanceMesh>("dragon.obj");
 
         auto material = components::Material();
         material.set_texture_slot_path(TextureType::Albedo, "oldiron/OldIron01_1K_BaseColor.png");
         material.set_texture_slot_path(TextureType::Normal, "oldiron/OldIron01_1K_Normal.png");
-        material.set_texture_slot_path(TextureType::Metallic, "whiteTexture");
-        material.set_texture_slot_path(TextureType::Roughness, "whiteTexture");
-        material.set_texture_slot_path(TextureType::Occlusion, "whiteTexture");
+        material.set_texture_slot_path(TextureType::Metallic, "normal_tiles_1k.png");
+        material.set_texture_slot_path(TextureType::Roughness, "normal_tiles_1k.png");
+        material.set_texture_slot_path(TextureType::Occlusion, "normal_tiles_1k.png");
         cubeObj.add_component<components::Material>(material);
     }
     {
         auto cubeObj = scene.create_game_object("loaded_dragon");
         cubeObj.get_component<components::Transform>().set_position(glm::vec3(-5.0f + 5, 1.0f, -10.0f - 5));
         cubeObj.get_component<components::Transform>().set_scale(glm::vec3(3, 3, 3));
         cubeObj.get_component<components::Transform>().set_rotation_euler(glm::vec3(0, 240, 0));
-        cubeObj.add_component<InstanceMesh>("dragon.obj");
+        cubeObj.add_component<components::InstanceMesh>("dragon.obj");
 
         auto material = components::Material();
         material.set_texture_slot_path(TextureType::Albedo, "oldiron/OldIron01_1K_BaseColor.png");
         material.set_texture_slot_path(TextureType::Normal, "oldiron/OldIron01_1K_Normal.png");
-        material.set_texture_slot_path(TextureType::Metallic, "whiteTexture");
-        material.set_texture_slot_path(TextureType::Roughness, "whiteTexture");
-        material.set_texture_slot_path(TextureType::Occlusion, "whiteTexture");
+        material.set_texture_slot_path(TextureType::Metallic, "normal_tiles_1k.png");
+        material.set_texture_slot_path(TextureType::Roughness, "normal_tiles_1k.png");
+        material.set_texture_slot_path(TextureType::Occlusion, "normal_tiles_1k.png");
```

Comment:
does serialisation not support using generated textures? does it have to use files being loaded?

Reply:
It does and is now fixed to do so!